### PR TITLE
Add a zone-scoped clock and features from the old clock package

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+
+* A `Clock.stopwatch()` method has been added that creates a `Stopwatch` that
+  uses the clock as its source of time.
+
+[`fake_async`]: https://pub.dartlang.org/packages/fake_async
+
+### Changes Relative to `clock` 0.1
+
+* The top-level `new` getter and `getStopwatch()` methods are deprecated.
+  `clock.new()` and `clock.getStopwatch()` should be used instead.
```
Comment (natebosch) [inline CHANGELOG.md]: Should this be stopwatch instead of getStopwatch?

File: lib/clock.dart [hunk 2]
```diff
@@ -12,164 +12,21 @@
...
-/// This class is designed with testability in mind. The current point in time
-/// (or [now()]) is defined by a function that returns a [DateTime]. By
-/// supplying your own time function or using [new Clock.fixed], you can control
-/// exactly what time a [Clock] returns and base your test expectations on that.
-class Clock {
-  /// The function that's called to determine this clock's notion of the current
-  /// time.
-  final DateTime Function() _time;
+/// Returns the current time as reported by [clock].
+@Deprecated("Use clock.now() instead.")
```
Comment (natebosch) [inline lib/clock.dart]: Do we want to release < 1.0.0 so we can clean these up first or are we going to leave the deprecated stuff in for a bit?

**Review by natebosch — APPROVED** [review]
Lgtm once Travis is green

File: CHANGELOG.md (hunk 1 above)
Comment (nex3) [reply CHANGELOG.md]: Done.

File: lib/clock.dart (hunk 2 above)
Comment (nex3) [reply lib/clock.dart]: I think it's fine to release as 1.0.0. I generally think of deprecations as meaning "we'll remove these next time we need to do a breaking release for some other reason" unless they're particularly difficult to maintain or they're blocking other better APIs.

**Pull request merged**